validator.go

```diff
@@ -117,13 +117,22 @@ func ValidateAll(xmlReader io.Reader) []error {
 
 // bufio implements a ByteReader but we explicitly don't want any buffering
 type byteReader struct {
-	r io.Reader
+	r   io.Reader
```

Comment:
Can you remove the extra space here? I think it was introduced previously when you added a field, but now it's no longer relevant to the change.

Otherwise looks great now!